### PR TITLE
Add JavaScript style guide, linter and formatter

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Add standardJS and resolve all identify issues.

#### Ticket
[Add JavaScript style guide, linter and formatter](https://dsdmoj.atlassian.net/browse/CBO-1612)

#### Why
To check JS code for programmatic and stylistic errors and enforce a JS coding standard throughout the codebase

### CODE REVIEW

File: .git-hooks/git-crypt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+printf 'Checking for git-crypt warnings...'
+
+test -d .git-crypt && git-crypt status &>/dev/null
+if [[ $? -ne 0 ]]; then
+  printf "warnings found!"
+  git-crypt status -e
+  exit 1
+else
+  printf "none found!"
+fi
```
Comment (jsugarman) [inline .git-hooks/git-crypt]: Does this not remove this test from the pre-commit hook. if you need to add something i would add the huskey stuff to the existing pre-commit hook.

File: app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.GraduatedPrice.js [hunk 2]
```diff
@@ -156,7 +156,7 @@
 
     responseErrored: function (response) {
       return Boolean(
-        response.hasOwnProperty('responseJSON') &&
+        Object.prototype.hasOwnProperty.call(response, 'responseJSON') &&
```
Comment (jsugarman) [inline app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.GraduatedPrice.js:159]: can't say i understand the differences here but has this been checked that it breaks nothing. The cucumber tests do test some fee calc functionality but can't say I would be confident this particular code would be covered by them

File: app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.UnitPrice.js [hunk 3]
```diff
@@ -207,7 +207,7 @@
           self.displayHelp(context, true)
         })
         .fail(function (response) {
-          if (response.hasOwnProperty('responseJSON') && response.responseJSON.errors[0] !== 'insufficient_data') {
+          if (Object.prototype.hasOwnProperty.call(response, 'responseJSON') && response.responseJSON.errors[0] !== 'insufficient_data') {
```
Comment (jsugarman) [inline app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.UnitPrice.js:210]: [ditto](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3628/commits/93cbb848cec8a454725c4e9df6154df349d2afca#r538439910)

**Review by jsugarman — CHANGES_REQUESTED** [review]
All the linting looks fine to me. one question on the protptype changes to fee calculator js.

Will need to change the pre-commit hook so we are still checking secret exposure on commit as far as i can tell - but have not tried

**Comment by jsugarman** [comment]
oh and 👍  for the separate commits - made it much easier to review

File: app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.GraduatedPrice.js (hunk 2 above)
Comment (naseberry) [reply app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.GraduatedPrice.js:159]: > can't say i understand the differences here but has this been checked that it breaks nothing. The cucumber tests do test some fee calc functionality but can't say I would be confident this particular code would be covered by them

In short, it prevents call object.prototype methods directly from an object, see: https://eslint.org/docs/rules/no-prototype-builtins.

This has been tested locally and will be tested once deployed to an env.

File: app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.UnitPrice.js (hunk 3 above)
Comment (naseberry) [reply app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.UnitPrice.js:210]: > [ditto](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3628/commits/93cbb848cec8a454725c4e9df6154df349d2afca#r538439910)

As above.

**Comment by naseberry** [comment]
> All the linting looks fine to me. one question on the protptype changes to fee calculator js.
> 
> Will need to change the pre-commit hook so we are still checking secret exposure on commit as far as i can tell - but have not tried

The changes were running both git-crypt and linting on the git pre-commit hook, however, I have rewritten this. It has no additional dependencies (Husky) and the shell script now includes linting.

**Comment by naseberry** [comment]
Sanity checked on dev-lgfs

**Review by jsugarman — APPROVED** [review]
👍  LGTM

**Pull request merged**